tiledb/sm/cpp_api/ndrectangle.h

```diff
@@ -269,6 +269,21 @@ class NDRectangle {
   std::shared_ptr<tiledb_ndrectangle_t> ptr() const {
     return ndrect_;
   }
+  /**
+   * Get the data type of the range at idx
+   *
+   * @param dim_idx The dimension index.
+   * @return The datatype of the range.
+   */
+  tiledb_datatype_t range_dtype(unsigned dim_idx) {
```

Comment:
There are other accessors that double up everything in terms of dimension _index_ and _name_.  Could we have an accessor that uses name too as I wrote the rest of my implementation in terms of the dimension name.  (Yes, I know, I could wrap this but while we are at making this complete may as well make it fully complete.)